### PR TITLE
Pull project repository before generating CaseStudy

### PR DESCRIPTION
**Pull request by Sinerum**
`vara-cs gen` now pulls before generating a case study, can be disabled with a `--no-update` flag.
Fixes se-sic/VaRA#847

### CODE REVIEW

File: varats-core/varats/utils/git_util.py [hunk 1]
```diff
@@ -572,6 +572,11 @@ def map_commits(
 )
 
 
+def update_repo(git_path: Path) -> None:
+    with local.cwd(git_path):
+        git('pull')
```
Comment (boehmseb) [inline varats-core/varats/utils/git_util.py]: Wouldn't it be cleaner to call `fetch` on the project's primary source?

**Review by boehmseb — CHANGES_REQUESTED** [review]

File: varats-core/varats/utils/git_util.py (hunk 1 above)
Comment (vulder) [reply varats-core/varats/utils/git_util.py]: Depends, I think select_latest picks from master (or the current branch?)
but, anyhow we probably need to update the branch with pull and not only fetch it

Comment (vulder) [inline varats-core/varats/utils/git_util.py]: I think, we should use the `pull_current_branch` implemented for the vara manager to not double the impl here.
The `pull_current_branch` does also work well when working with multiple repos in parallel

File: varats/varats/tools/driver_casestudy.py [hunk 2]
```diff
@@ -156,10 +156,14 @@ def __casestudy_status(
     is_flag=True,
     help="Ignore revisions that are marked as blocked."
 )
+@click.option(
+    "--no-update", is_flag=True, help="Project repository will not be updated."
+)
```
Comment (vulder) [inline varats/varats/tools/driver_casestudy.py]: a) which default do we want? update or no update?
b) we should invert the flag and set it by default to false, makes the code easier to understand

**Review by vulder — CHANGES_REQUESTED** [review]

File: varats-core/varats/utils/git_util.py (hunk 1 above)
Comment (Sinerum) [reply varats-core/varats/utils/git_util.py]: It uses the current branch.

Comment (vulder) [reply varats-core/varats/utils/git_util.py]: I thought so. This means pull is better than just fetch.

**Review by vulder — APPROVED** [review]

**Review by boehmseb — APPROVED** [review]

**Pull request merged**